rewriter/Minitest.cc

```diff
@@ -145,6 +143,90 @@ string to_s(core::Context ctx, unique_ptr<ast::Expression> &arg) {
     return arg->toString(ctx);
 }
 
+// This returns `true` for expressions which can be moved from class to method scope without changing their meaning, and
+// `false` otherwise. This mostly encompasses literals (arrays, hashes, basic literals), constants, and sends that only
+// involve the other things described.
+bool canMoveIntoMethodDef(const unique_ptr<ast::Expression> &exp) {
+    if (ast::isa_tree<ast::Literal>(exp.get())) {
+        return true;
+    } else if (auto list = ast::cast_tree<ast::Array>(exp.get())) {
+        return absl::c_all_of(list->elems, [](auto &elem) { return canMoveIntoMethodDef(elem); });
+    } else if (auto hash = ast::cast_tree<ast::Hash>(exp.get())) {
+        return absl::c_all_of(hash->keys, [](auto &elem) { return canMoveIntoMethodDef(elem); }) &&
+               absl::c_all_of(hash->values, [](auto &elem) { return canMoveIntoMethodDef(elem); });
+    } else if (auto send = ast::cast_tree<ast::Send>(exp.get())) {
+        return canMoveIntoMethodDef(send->recv) &&
+               absl::c_all_of(send->args, [](auto &elem) { return canMoveIntoMethodDef(elem); });
```

Comment:
This is a neat trick. I was thinking that you'd have to look at `send->fun` but in fact you can just look at whether the receiver would change meaning.